tests/strace-n.c

```diff
@@ -0,0 +1,42 @@
+/*
+ * Test strace's -n option.
+ *
+ * Copyright (c) 2020 The strace developers.
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier: GPL-2.0-or-later
+ */
+
+#include "tests.h"
+#include "scno.h"
+
+#if defined __NR_socketcall && defined __NR_listen && defined __i386__
+
+# include <stdio.h>
+# include <unistd.h>
+# include <errno.h>
+
+# define SYS_socket_subcall 500
+# define SC_listen 4
+
+int
+main(void)
+{
+	if (syscall(__NR_socketcall, 0L, 0L, 0L, 0L, 0L) < 0
+		&& EINVAL == errno)
+	{
+		const long args[] = { 0, 0 };
+		syscall(__NR_socketcall, SC_listen, args);
+		printf("[%4u] listen(0, 0) = -1 ENOTSOCK (%m)\n", SYS_socket_subcall + SC_listen);
```

Comment:
```SYS_socket_subcall + SC_listen``` is not a real syscall number, it's something from strace internals.
I suggest printing the real syscall number being used, i.e. ```__NR_socketcall```.

Also, please do not hardcode a particular error code to avoid potential issues with seccomp filters.